pkg/engine/lifecycletest/refresh_test.go

```diff
@@ -775,7 +775,7 @@ func TestCanceledRefresh(t *testing.T) {
 	}
 
 	snap, err := op.RunWithContext(ctx, project, target, options, false, nil, validate)
-	assert.Error(t, err)
+	assert.ErrorContains(t, err, "BAIL: canceled")
```

Comment:
Strengthen the assertion here. Just checking that _an error_ happened is generally too weak a test assertion.